pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1592,21 +1609,71 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 			}
 			// increment potential IP
 			potentialDefaultIP = incrementIPForDefaultCIDR(potentialDefaultIP)
-
-			// check for overlap
-			// if it overlaps continue with the loop
+			//check for overlap
+			// if it overlaps continue
+			//no need for boolean check as if cluster vpc fails it wont get this far
```

Comment:
Nitpick - I think we can remove this comment, the checks are self explanatory and called out in the comment above the loop